src/JSInterop/Microsoft.JSInterop/src/PublicAPI.Unshipped.txt

```diff
@@ -36,7 +36,7 @@ static Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(this Microsoft.JS
 static Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(this Microsoft.JSInterop.IJSRuntime! jsRuntime, string! identifier, System.TimeSpan timeout, params object?[]? args) -> System.Threading.Tasks.ValueTask
 *REMOVED*static Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(this Microsoft.JSInterop.IJSRuntime! jsRuntime, string! identifier, params object![]! args) -> System.Threading.Tasks.ValueTask
 static Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(this Microsoft.JSInterop.IJSRuntime! jsRuntime, string! identifier, params object?[]? args) -> System.Threading.Tasks.ValueTask
-virtual Microsoft.JSInterop.JSRuntime.ReadJSDataAsStreamAsync(Microsoft.JSInterop.IJSStreamReference! jsStreamReference, long totalLength, long maxBufferSize, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<System.IO.Stream!>!
+virtual Microsoft.JSInterop.JSRuntime.ReadJSDataAsStreamAsync(Microsoft.JSInterop.IJSStreamReference! jsStreamReference, long totalLength, long pauseIncomingBytesThreshold, long resumeIncomingBytesThreshold, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<System.IO.Stream!>!
```

Comment:
I'm not a big fan of this API. 

It exposes a bunch of internal details and it does so in a way that is problematic and that can make it easy to introduce security issues if you are not careful. The defaults need to be safe (the default can't never be "unlimited" on a server app) and we should add a remark on the comments about the security implications of these parameters.

Reply:
@javiercn Would you consider it sufficient to:

* have some default like 100KB or so
* remove the behavior of "zero means unlimited" so that if someone really wants unlimited they have to pass `int.MaxValue` or similar
* include a mention in the XML docs for the parameter saying something like `Avoid specifying an excessively large value because this could allow clients to exhaust server memory.`

?

Reply:
That sounds about right.

We would need to figure out a good default. It's a trade-off between speed and availability I think

Reply:
To clarify this here, we _do_ have a safe default ([indicated by `-1` - zero is _not_ the default](https://github.com/dotnet/aspnetcore/pull/33916/files#diff-43313bc99747b2864849d5c668e74ea4d2a49ef8cb427a7669aa299a7104cd8eR29)). Users should be working with the `JSStreamReference` and _not_ the  `JSRuntime` methods seen here. This follows the exact pattern seen in the `PipeOptions`:

Documentation: https://github.com/dotnet/aspnetcore/pull/33491#discussion_r657754429

Default Args:
https://github.com/dotnet/runtime/blob/9fb6e818cda2e4364e085fe0b8ecebed97db9596/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs#L30-L31

What the safe defaults translate to:
https://github.com/dotnet/runtime/blob/9fb6e818cda2e4364e085fe0b8ecebed97db9596/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs#L47-L69

(65k and 32.5k bytes). Originally we were using a 100kb but per https://github.com/dotnet/aspnetcore/pull/33491#discussion_r657754429, it seemed like using the `Pipe` defaults was preferred.

Are we okay leaving this as is in this case?

Reply:
Would 7785751945 resolve these concerns?

Reply:
/ping